watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteFragment.java

```diff
@@ -246,4 +302,28 @@ public void onPause() {
         super.onDestroy();
         if (mMenu != null) closeMenu();
     }
+
+    private void markSubscribed(ArrayList<Site> sites) {
+        for (Site site : sites) {
+            if (site.getId().equals(mSite.getId())) {
+                FloatingActionButton subscribeButton = (FloatingActionButton) mView.findViewById(R.id.site_subscribe_site);
+                subscribeButton.setTitle("Unsubscribe from Site");
+                subscribeButton.setIcon(R.drawable.ic_bookmark_white_36dp);
+                mSubscribed = true;
+            }
+        }
+    }
```

Comment:
Wouldn't it be better to just check if the user is subscribed to this particular site?